zilliqa/src/consensus.rs

```diff
@@ -2808,4 +2867,56 @@ impl Consensus {
     ) -> Result<()> {
         self.block_store.report_outgoing_message_failure(failure)
     }
+
+    pub fn tick(&mut self) -> Result<()> {
```

Comment:
Because I wasn't sure if you wanted to regard a process normally executed periodically as a "timeout" .

Necessity, really - we generate traffic when we trigger block requests; I couldn't think of a way for a message to trigger these without risking generating either too much traffic (and blocking the node - and others - from other work, which was my initial problem), or too little traffic (and never requesting the blocks it needed to complete its view of the chain). So, trigger on a tick ..